src/cmap/connection.ts

```diff
@@ -369,7 +372,28 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
 
     // always emit the message, in case we are streaming
     this.emit('message', message);
-    const operationDescription = this[kQueue].get(message.responseTo);
+    let operationDescription = this[kQueue].get(message.responseTo);
+
+    if (!operationDescription && this.isMonitoringConnection) {
```

Comment:
What if it's a monitoring connection that doesn't support streaming responses?  Should we still enter this code block?  If not, is there a way to check for that condition?

Reply:
This code block should never get entered for a non-streaming monitoring connection as the operation description must always be there in that case.

Reply:
True, but you could say the same about a non-monitoring connection. Would you want just:

```
if (!operationDescription)
```

Reply:
Theoretically we could, but I wanted the scope of this fix just to be limited to monitoring connections and handle the entire connection refactor with NODE-4820.

Reply:
What Durran said, this will become a non-issue after NODE-4820, and if we do have any unexpected behavior with missing operation descriptions on regular operations, that would be a better time to change the behavior. I'm not sure if we have an easy way of differentiating streaming monitoring connections from non-streaming (if we do, then I'm not opposed to adding an extra guard here). But keeping it for all monitoring connections seems low risk, since the worst case scenario is we hand a hello response to a hello handler that's detached from the original request.